appinventor/blocklyeditor/src/warningHandler.js

```diff
@@ -386,10 +386,20 @@ Blockly.WarningHandler.prototype["checkIsNotInLoop"] = function(block) {
   }
 };
 
-Blockly_loopBlockTypes =
-  // add more later
-  ["controls_forEach", "controls_forRange", "controls_while"] ;
-
+// TODO: Maybe change this to a property of the block, instead of maintaining
```

Comment:
This seems like it would be a straightforward thing to add, i.e. have a `{?boolean=}` property of the block `isLoopBlock`. It would be faster than looking up the block's type in an array. Maybe in another PR?